test/python/transpiler/test_preset_passmanagers.py

```diff
@@ -940,3 +941,15 @@ def test_size_optimization(self, level):
 
         self.assertLess(circ.size(), qc.size())
         self.assertLessEqual(circ.depth(), qc.depth())
+
+
+@ddt
+class TestGeenratePresetPassManagers(QiskitTestCase):
+    """Test generate_preset_pass_manager function."""
+
+    @data(0, 1, 2, 3)
+    def test_with_backend(self, optimization_level):
```

Comment:
Do we want to add a test where the backend is `None` so we cover both cases of the if-clause in the function?

Reply:
Yep, that's a good idea. Done in: https://github.com/Qiskit/qiskit-terra/pull/8125/commits/c40a0f127f2f69384ee852824029b9db1feba88e